src/components/organisms/error-boundary/index.tsx

```diff
@@ -0,0 +1,164 @@
+})
+
+class ErrorBoundary extends React.Component<Props, State> {
+  public state: State = {
+    hasError: false,
+  }
+
+  static getDerivedStateFromError(error: Error) {
+    const state: State = {
+      hasError: true,
+      status: undefined,
+      message: error.message,
+    }
+
+    if (isNetworkError(error)) {
+      state.status = error.response?.status
+    }
+
+    return state
+  }
+
+  public async componentDidCatch(error: Error, errorInfo: ErrorInfo) {
+    const shouldTrack = await shouldTrackEvent(error)
+
+    if (!shouldTrack) {
+      return
+    }
+
+    const properties = getTrackingInfo(error, errorInfo)
+    analytics.track("error", properties)
+  }
+
+  public dismissError = () => {
+    navigate(-1)
+    this.setState({ hasError: false })
+  }
+
+  public render() {
+    if (this.state.hasError) {
+      return (
+        <div className="h-screen flex items-center justify-center px-large">
+          <div className="max-w-[600px]">
+            <div>
+              <div>
+                {this.state.status && (
+                  <p className="text-grey-60 opacity-75 inter-small-semibold">
+                    {this.state.status}
+                  </p>
+                )}
+                <h1 className="inter-xlarge-semibold mb-xsmall">
+                  {errorMessage(this.state.status)}
+                </h1>
+                <p className="inter-base-regular text-grey-50">
+                  {errorDescription(this.state.status)}
+                </p>
+              </div>
+
+              <div className="w-full flex items-center  mt-xlarge">
+                <Button
+                  size="small"
+                  variant="primary"
+                  onClick={this.dismissError}
+                >
+                  Back to dashboard
+                </Button>
+              </div>
+            </div>
+          </div>
+          <div></div>
+        </div>
+      )
+    }
+
+    return this.props.children
+  }
+}
+
+const isNetworkError = (error: Error): error is AxiosError => {
+  return error["response"] !== undefined && error["response"] !== null
+}
+
+const shouldTrackEvent = async (error: Error) => {
+  // Don't track errors in development
+  if (!IS_PROD) {
+    return false
+  }
+
+  // Don't track 404s
+  if (isNetworkError(error) && error.response?.status === 404) {
+    return false
+  }
+
+  const res = await getAnalyticsConfig().catch(() => undefined)
+
+  // Don't track if we have no config to ensure we have permission
+  if (!res) {
+    return false
+  }
+
+  // Don't track if user has opted out from sharing usage insights
+  if (res.analytics_config.opt_out) {
+    return false
+  }
+
+  return true
+}
+
+const errorMessage = (status?: number) => {
+  if (!status) {
+    return "An unknown error occured"
+  }
+
+  return {
+    401: "You are not logged in",
+    403: "You do not have permission perform this action",
+    404: "Page was not found",
+    500: "An unknown server error occured",
```

Comment:
nit(non-blocking): should we have a default value for these rather than them being undefined if they fail?